tests/test_efficient_semivariance.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+
+from pypfopt import risk_models, EfficientFrontier
+from pypfopt.efficient_semivariance import EfficientSemivariance
+from tests.utilities_for_tests import setup_efficient_semivariance
+
+
+def test_efficient_return():
+    es = setup_efficient_semivariance()
+    w = es.efficient_return(0.25 / 252)
+    assert isinstance(w, dict)
+    assert set(w.keys()) == set(es.tickers)
+    np.testing.assert_almost_equal(es.weights.sum(), 1)
+    assert all([i >= -1e-5 for i in w.values()])
+
+    np.testing.assert_allclose(
+        es.portfolio_performance(),
+        (0.2500001777470832, 0.10035962239578998, 2.291760094911752),
+    )
+
+
+def test_efficient_semivariance_vs_heuristic():
+    benchmark = 0
+    es = setup_efficient_semivariance()
+    es.efficient_return(0.20 / 252)
+    mu_es, semi_deviation, _ = es.portfolio_performance()
+
+    mean_return, historic_returns = setup_efficient_semivariance(data_only=True)
+
+    pairwise_semivariance = risk_models.semicovariance(
+        historic_returns, returns_data=True, benchmark=0, frequency=1
+    )
+    ef = EfficientFrontier(mean_return, pairwise_semivariance)
+    ef.efficient_return(0.20 / 252)
+    mu_ef, _, _ = ef.portfolio_performance()
+    mu_ef *= 252
+    portfolio_returns = historic_returns @ ef.weights
+    drops = np.fmin(portfolio_returns - benchmark, 0)
+    T = historic_returns.shape[0]
+    semivariance = np.sum(np.square(drops)) / T * 252
+    semi_deviation_ef = np.sqrt(semivariance)
+
+    assert semi_deviation < semi_deviation_ef
```

Comment:
For this sample, the difference in the semi-deviation is quite small (`0.0916` vs. `0.0917`). For the weekly test below, the difference is more significant (`0.0823` vs. `0.0928`).